elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -82,11 +80,24 @@ public SQLQueryEngine(MetaDataStore metaDataStore,
     public SQLQueryEngine(MetaDataStore metaDataStore,
                     com.yahoo.elide.contrib.dynamicconfighelpers.compile.ConnectionDetails defaultConnectionDetails,
                     Map<String, com.yahoo.elide.contrib.dynamicconfighelpers.compile.ConnectionDetails> detailsMap) {
-        this(metaDataStore, defaultConnectionDetails);
-        detailsMap.forEach((name, details) -> {
-            this.connectionDetailsMap.put(name, new ConnectionDetails(details.getDataSource(),
-                            SQLDialectFactory.getDialect(details.getDialect())));
-        });
+        if (defaultConnectionDetails == null) {
+            this.defaultConnectionDetails = null;
+        } else {
+            this.defaultConnectionDetails = new ConnectionDetails(defaultConnectionDetails.getDataSource(),
+                            SQLDialectFactory.getDialect(defaultConnectionDetails.getDialect()));
+        }
+        if (!(detailsMap == null || detailsMap.size() == 0)) {
+            detailsMap.forEach((name, details) -> {
+                this.connectionDetailsMap.put(name, new ConnectionDetails(details.getDataSource(),
+                                SQLDialectFactory.getDialect(details.getDialect())));
+            });
+        }
+        this.metaDataStore = metaDataStore;
```

Comment:
Is there a reason we moved all this initialization here (and removed final keywords as well from the parent class)?

Reply:
need to call construct table after initializing connection details map.